src/fs/file_info.go

```diff
@@ -46,13 +54,54 @@ func (fi *fileSorter) Less(i, j int) bool {
 	return strings.ToLower(fi.files[i].name) < strings.ToLower(fi.files[j].name)
 }
 
+func (f *fileCacheInfo) invalidateCache() {
+	f.status = false
+	f.size = 0
+	f.mtime = time.Time{}
+	f.endpoint = ""
+}
+
+func (f *fileCacheInfo) fillCache(filePath, share, path string) {
+	parentDir := filepath.Dir(filePath)
+	filename := filepath.Base(filePath)
+
+	thumbnailDirPath := filepath.Join(parentDir, ".fscache/thumbnails")
+	thumbnailPath := filepath.Join(thumbnailDirPath, filename)
+	thumbnailInfo, err := os.Stat(thumbnailPath)
+	if os.IsNotExist(err) {
+		f.status = false
+	} else {
+		f.status = true
+		f.size = thumbnailInfo.Size()
+		f.mtime = thumbnailInfo.ModTime()
+		if len(path) == 0 {
+			path = "/"
+		}
+		path := filepath.Join(path, filename)
+		log("inside fill cache. Path is", path)
+		f.endpoint = fmt.Sprintf(`/cache?s=%s&p=%s`, share, path)
```

Comment:
I believe relative URL like this would be bad API design, we have two options:
1) Make it absolute (will need to check if the correct Host header is returned when connecting via proxy)
2) Just send the name of the thumbnail file if it exists else nil, and the client will prepare the complete URL just as in the case of accessing any file on a share.

@cpg What do you think?